code/data_structures/src/list/singly_linked_list/operations/find/search_element_python.py

```diff
@@ -0,0 +1,103 @@
+class Node:
+    '''
+    A Node in a singly-linked list
+
+    Parameters
+    -------------------
+    data :
+        The data to be stored in the node
+    next:
+        The link to the next node in the singly-linked list
+    '''
+
+    def __init__(self, data=None, next=None):
+        self.data = data
+        self.next = next
+
+    def __repr__(self):
+        ''' Node representation as required'''
+        return (self.data)
+
+class NodeAdapter1(Node):
+    def __bool__(self):
+        return False
+
+class NodeAdapter2(Node):
+    def __bool__(self):
+        return True
+    
+class SinglyLinkedList:
+    def __init__(self,NodeType):
+        self.head = None
+        self._NodeType = NodeType
+
+    def append(self, data):
+        '''
+        Inserts New data at the ending of the Linked List
+        Takes O(n) time
+        '''
+        new_node = _NodeType(data = data)
+        if self.head is None:
+            self.head = new_node
+            return
+
+        current_node = self.head
+        while current_node.next is not None:
+            current_node = current_node.next
+
+        current_node.next = new_node
+
+    def find(self, data):
+        '''
+        Returns the first index at which the element is found
+        Returns -1 if not found
+        Takes O(n) time
+        '''
+
+        if self.head is not None:
+            current_node = self.head
+            index = 0
+            while current_node is not None:
+                if current_node.data == data:
+                    return index
+                current_node = current_node.next
+                index += 1
+        return -1
```

Comment:
Thanks for adoption of my suggestion.
In addition, I prefer use only one return statement or place it on bottom:
```
def find(self, data):
    # initial
    current_node = self.head
    index = 0

    # searching
    while current_node is not None and current_node.data != data:
        current_node = current_node.next
        index += 1

    if current_node is None:
        return -1
    else:
        return index
```